src/store.ts

```diff
@@ -187,6 +188,16 @@ const store: TypedVuexStore = new TypedVuexStore({
     setSawNewFeature(state: VuexStoreState, seen: boolean) {
       state.onboardingData.sawNewFeature = seen;
     },
+    setSawGiveaway(state: VuexStoreState, seen: boolean) {
+      const currentDate = new Date();
+      const cutoffDate = new Date('2024-10-30'); // giveaway ending date
+
+      if (currentDate >= cutoffDate) {
+        state.onboardingData.sawGiveaway = true; // if the current date is past the giveaway date, then we set saw giveaway to true, meaning the modal won't be visible to them
+      } else {
+        state.onboardingData.sawGiveaway = seen; // otherwise, if the current date is within the giveaway time period, then whether the modal is visible or not is dependent on whether they've seen it before
+      }
+    },
```

Comment:
Should we also display a confirmation text to make sure the user knows that they submitted the giveaway properly?